### ISSUE TITLE
UnitTests: Error in Frontend/Basic.t

### ISSUE THREAD

**Issue by bschmalhofer** [RotherOSS/otobo#353]
Looks like this is reproducible. Needs to be investigated.

```
scripts/test/Frontend/Basic.t                                                                  (Wstat: 0 Tests: 858 Failed: 8)
  Failed tests:  542, 545, 548, 619, 631, 799, 807, 833

```

**Comment by bschmalhofer** [comment]
Confirmed, same errors under 10_1 as on August 14th.
No errors on 10_0.
Also made a small fix so that the test numbers are consistent when there is a failure.

**Comment by bschmalhofer** [comment]
I have analysed this for the Frontend module `AgentAppointmentEdit`. For this module a fatal error should be thrown as quite understandably the ChallengeToken does not match. The HTTP response that should be generated in the case has:

- Status 200 OK
- Content-type HTML
- Content-Length > 0

For analysing the issue I considered two cases:

1. rel-10_0 running under Apache with _apache2-httpd-psgi.include.conf_
2. rel-10_1 running under Apache with  _apache2-httpd-psgi.include.conf_

In both cases otobo.psgi is handled by the PerlResponseHandler Plack::Handler::Apache2. 

The initial impression was that 10.0 was working while 10.1 was broken. But this is only partially true, as the test script _Frontend/Basic.t_ has limited knowledge what it should test.

For 10_0 we get a response:
```
HTTP/1.1 200 OK
Cache-Control: no-cache
Connection: close
Date: Fri, 14 Aug 2020 12:49:18 GMT
Pragma: no-cache
Server: Apache/2.4.41 (Ubuntu)
Content-Length: 2
Content-Type: application/json; charset=utf-8
Expires: Tue, 1 Jan 1980 12:00:00 GMT
Client-Date: Fri, 14 Aug 2020 12:49:18 GMT
Client-Peer: 127.0.0.1:80
Client-Response-Num: 1
Content-Disposition: inline; 
X-Frame-Options: SAMEORIGIN
X-UA-Compatible: IE=edge,chrome=1

[]
```
This is an empty JSON array, which is acceptable for _Frontend/Basic.t_ .

For t0_1 we get:

```
HTTP/1.1 200 OK
Connection: close
Date: Fri, 14 Aug 2020 12:42:51 GMT
Server: Apache/2.4.41 (Ubuntu)
Content-Length: 0
Client-Date: Fri, 14 Aug 2020 12:42:51 GMT
Client-Peer: 127.0.0.1:80
Client-Response-Num: 1
```
This empty message is not acceptable for _Frontend/Basic.t_ .

Both responses are incorrect as we really expect HTML with the error message **Invalid Challenge Token!**.

The root of the problem is that `Kernel::Output::HTML::Layout::FatalError()`calls `exit()` before the real response is delivered.This is a known problem as reported in #180. A solution exist in the 10_1 branch but this is currently only enabled for the InterfaceInstaller pages. See #323 

TODO:

- [x] backport the exception handling fix from 10.1 to 10.0

- [x] in 10.0: throw the exception for FatalError() and CustomerFatalError() when OTOBO_RUNS_UNDER_PSGI is set

- [x] test with 10.0

- [x] merge the 10.0 changes into 10.1

- [x] in 10.1: throw the exception for FatalError() and CustomerFatalError() when OTOBO_RUNS_UNDER_PSGI is set

- [x] test with 10.1

**Comment by bschmalhofer** [comment]
Looks good.  Closing the issue.
